nixos/modules/tasks/auto-upgrade.nix

```diff
@@ -91,24 +112,36 @@ let cfg = config.system.autoUpgrade; in
       path = with pkgs; [ coreutils gnutar xz.bin gzip gitMinimal config.nix.package.out ];
 
       script = let
-          nixos-rebuild = "${config.system.build.nixos-rebuild}/bin/nixos-rebuild";
-        in
+        nixos-rebuild = "${config.system.build.nixos-rebuild}/bin/nixos-rebuild";
+        date = "/run/current-system/sw/bin/date";
+        readlink = "/run/current-system/sw/bin/readlink";
+        shutdown = "/run/current-system/sw/bin/shutdown";
```

Comment:
I think it'd be better to make these paths relative to `pkgs.coreutils` and `pkgs.systemd`, not the current system.

Reply:
@Atemu: done.